### PR TITLE
decompressor: add config flag to ignore no-transform

### PR DESCRIPTION
**Pull request by jpsim**
Commit Message: Add config flag to decompress even when no-transform is specified
Additional Description: This allows matching the more lenient behavior of other client-side networking libraries such as OkHttp or URLSession, especially in cases where the remote server is not under the client developer's control.
Risk Level: Low, defaults are unaffected, change is opt-in.
Testing: Added unit tests for existing `no-transform` behavior (there was none) and a new test validating the new configuration flag.
Docs Changes: Updated.
Release Notes: Added.
Platform Specific Features:

### CODE REVIEW

File: api/envoy/extensions/filters/http/decompressor/v3/decompressor.proto
Comment (snowp) [inline api/envoy/extensions/filters/http/decompressor/v3/decompressor.proto:31]: Normally we'll use the primitive bool type when the default is false

File: source/extensions/filters/http/decompressor/decompressor_filter.h
Comment (snowp) [inline source/extensions/filters/http/decompressor/decompressor_filter.h:170]: const

**Review by snowp — CHANGES_REQUESTED** [review]
Couple nits, otherwise this lgtm!

**Review by snowp — APPROVED** [review]
LGTM! 

@rojkov wanna take a look as owner of this extension?

**Comment by snowp** [comment]
Also this needs @envoyproxy/api-shepherds approval

File: api/envoy/extensions/filters/http/decompressor/v3/decompressor.proto
Comment (htuch) [inline api/envoy/extensions/filters/http/decompressor/v3/decompressor.proto:29]:
```suggestion
    // If set to true, will decompress response even if a *no-transform* cache control header is set.
```

File: api/envoy/extensions/filters/http/decompressor/v3/decompressor.proto
Comment (htuch) [inline api/envoy/extensions/filters/http/decompressor/v3/decompressor.proto:30]: Defeault is implied by scalar bool so can be dropped.

**Review by htuch — COMMENTED** [review]
LGTM API modulo nits.

**Comment by jpsim** [comment]
@snowp @htuch feedback has been addressed and CI appears to be happy.

**Review by htuch — COMMENTED** [review]
/lgtm api

**Review by rojkov — APPROVED** [review]
Thanks! LGTM

**Comment by jpsim** [comment]
Thanks for the feedback everyone.

**Pull request merged**
